### DISCUSSION TITLE
Cannot install sentence-transformers to use sentence BERT

### DISCUSSION THREAD

**Discussion by Bianca858** [asreview/asreview#1945] [Q&A] [answered]
Good day all

I need some assistance please. I have been struggling for days.

I can install ASReview, upload the data files and choose a model. The model I chose was the following:
Feature extraction technique:                   Sentence BERT
Classifier:                                                   Support Vector Machine
Query strategy:                                          Maximum
Balance strategy:                                        Dynamic resampling (Double)

And then I cannot pass this stage, and I repeatedly get the following message: Install sentence-transformers package to use Sentence BERT.

When I try to do it, it simply won't work. I am using Windows.

Please help!

Regards
Bianca

**Comment by Rensvandeschoot** [comment]
Did you try:
```
pip install asreview[sentence_transformers] 
```
And if so, what is your exact error message?

**Comment by Bianca858** [comment]
Dear Rens

I hope you are well.


I tried pip install asreview[sentence_transformers]

And I received the following message on command prompt:

[A screen shot of a computer  Description automatically generated]

When I tried rerunning the model and it was warming up, I once again got the following error:

***@***.***

I am not sure what to do now.

Regards
Bianca

**Reply by timovdk** [reply]
Hi @Bianca858!
Thank you for your response! What version of Python are you using? You can check this by running `python --version` in the command prompt. If it is between 3.9... and 3.12... you can try to run the following line to install the sentence transformers: `python -m pip install asreview sentence-transformers tf-keras`.

**Comment by Bianca858** [comment]
Hi Timo

I have 3.13.1

Bianca

**Reply by timovdk** [reply, marked as answer]
In that case you have two options, because Pytorch (a dependency of `sentence-transformers`) does not have support Python 3.13 yet:
- Downgrade to Python 3.12, and install the toolchain with this command: `python -m pip install asreview sentence-transformers tf-keras`
- Install a nightly build (the very latest version of a package) of Pytorch using this command: `python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cpu` after which you can install the sentence transformers using the following command: `python -m pip install sentence-transformers tf-keras`.

However, if you want to use deep learning classifiers such as LSTM, and Neural Networks in the future, you would have to downgrade to Python 3.12, because these depend on tensorflow, of which even the nightly version does not support Python 3.13 yet.

Please let me know if anything is unclear!

**Comment by Bianca858** [comment]
I downloaded 3.12 and its working!!!

I received this message though, is it anything I need to worry about:

2024-12-12 19:55:55.981865: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-12-12 19:55:57.223046: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
WARNING:tensorflow:From C:\Users\01470012\AppData\Local\Programs\Python\Python312\Lib\site-packages\tf_keras\src\losses.py:2976: The name tf.losses.sparse_softmax_cross_entropy is deprecated. Please use tf.compat.v1.losses.sparse_softmax_cross_entropy instead.

**Reply by timovdk** [reply]
Hi @Bianca858,
Great to hear! You do not have to worry about these warnings.
